sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -248,12 +248,17 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
    * `HadoopFsRelation` node(s) as part of its logical plan.
    */
   def recacheByPath(spark: SparkSession, resourcePath: String): Unit = {
```

Comment:
how about we change the method as
```
def recacheByPath(spark: SparkSession, resourcePath: String, options: Map[String, String]=Map.empty)
```
so that we can avoid the new method below?

Reply:
That also works, but it feels a little bit weird to couple the data source options concept with the cache manager...